### PR TITLE
[RFC] basefiles,procd: start process on reload [if not running]

### PR DESCRIPTION
**Pull request by commodo**
@dedeckeh @nbd168 

This is a continuation on change 6713694
That one changed the semantic from not doing restart on reload fail,
to defining default reload as restart.

But there is also the case where, if a service is not
running, reload could start it.

Of course, a reload hook could also implement this itself.
But, it feels like this would be a nice feature to have
for procd [able] services.

Still, if a service init script wants/needs to define
it's own `service_running` script, it can do that,
and it will be used instead of relying on procd [by default].

There are 2 base packages (system & qos-scripts) who's start & reload
calls do the same thing.
And it seems that for daemon-less init scripts, reload and start
do the same thing.

Signed-off-by: Alexandru Ardelean <ardeleanalex@gmail.com>

### CODE REVIEW

File: package/system/procd/files/procd.sh [hunk 1]
```diff
@@ -413,6 +413,33 @@ _procd_set_config_changed() {
...
+	json_init
+	json_add_string name "$service"
+	local result="$(ubus call service list "$(json_dump)")"
+
+	# load & process result
+	json_load "$result"
+
+	# silence warnings from here on
+	local _json_no_warning=1
+	result=1
```
Comment (blogic) [inline package/system/procd/files/procd.sh]: dont reuse the result var please it makes the code hard to read. also drop the default value assignment.

File: package/system/procd/files/procd.sh [hunk 2]
```diff
@@ -413,6 +413,33 @@ _procd_set_config_changed() {
...
+	json_select "$service" && \
+		json_select "instances" && \
+		json_select "instance1" && \
+		json_get_var result running && \
+		[ "$result" == "1" ] && \
```
Comment (blogic) [inline package/system/procd/files/procd.sh]: drop this from the && chain and call it unconditionally.

File: package/system/procd/files/procd.sh (hunk 1 above)
Comment (commodo) [inline package/system/procd/files/procd.sh]: ack

File: package/system/procd/files/procd.sh (hunk 2 above)
Comment (commodo) [inline package/system/procd/files/procd.sh]: ack

**Comment by blogic** [comment]
i am not sure if this approach is the right one. in theory when sending a service signature to procd it should compare it to the existing one and then start it if changed and/or not running. if it does not do that, then we should fix it inside procd rather than add a shell wrapper. i'll look into this next week when i am near test hw again.

**Comment by commodo** [comment]
I think the reload hook [if implemented by the service] does not get passed to procd.
It can be whatever you want it to be [ ex. ubus call ].

so, if a service is not running, it will just non-zero.
the discussion is more like: when calling a service reload, do we want the init.d logic to start the service ? or just let the reload call fail ?

**Comment by blogic** [comment]
reload should start none running processes.

**Comment by commodo** [comment]
understood ;
let's do it like this: i'll split up some of the commits from this PR into smaller PRs [ some are easy to agree upon ]
i'll re-spin the others based on comments/discussion here

**Pull request closed without merging**